mesa/experimental/cell_space/voronoi.py

```diff
@@ -0,0 +1,58 @@
+from collections.abc import Sequence
+from itertools import combinations
+from random import Random
+from typing import Optional
+
+from pyhull.delaunay import DelaunayTri
+
+from mesa.experimental.cell_space.cell import Cell
+from mesa.experimental.cell_space.discrete_space import DiscreteSpace
+
+
+class VoronoiGrid(DiscreteSpace):
+    def __init__(
+        self,
+        centroids_coordinates: Sequence[Sequence[float]],
```

Comment:
Would this work with Shapely [Points](https://shapely.readthedocs.io/en/stable/reference/shapely.Point.html)? That might be very useful for Mesa geo applications (cc @wang-boyu).